Awesome Books Project

peterdgreat opened the pull request:
This is an implementation of the javascript add and remove from a list item when the button is clicked. This also preserves data on the local storage.
This is a plain Html with no stylings yet.

wikitelmex1 committed:
local storage on work

- add element to array and to html
- remove element on array and html
- starting working with local storage crud

wikitelmex1 committed:
Update script.js

- add localstorage crud
- preserve and load data

Commit:
README updated

Commit:
fixing linters

Commit:
linters -fixed

Commit:
linters -fixes

wikitelmex1 committed:
improvements

- add live demo
- html improvements
- index id randomly generated to avoid coalitions
- fixed local storage and collection issues

wikitelmex1 committed:
Update script.js

- fixed var types error

Davidosky007 commented on script.js:
```diff
@@ -0,0 +1,85 @@
+  const isIndex = (element) => element.index === idIndex.toString();
+
+  books.splice(books.findIndex(isIndex), 1);
+  booksTable.removeChild(tr);
```
* I get this error when i click the remove button for the first book created kindly fix this error in the console 
![image](https://user-images.githubusercontent.com/52497246/133297636-6a55af59-a38c-4aa2-bba3-6e15f9c7f57d.png)

Wikitelmex replied:
You are right, it happens with the elements that have an ID that starts with Zero, for example "0371850523373020", when call the function onClick, its called onClick ="RemoveRowBooks(0371850523373020)", that automatically deleted the starting zero, we just added "RemoveRowBooks('0371850523373020')" single quotations to avoid that.

Davidosky007 reviewed (changes requested):
## Hi @Wikitelmex @peterdgreat 👋, 

Good job so far! 👍 
* Nice looking repo
* descriptive commit messges 
* professional Readme file 
* Well written PR 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

wikitelmex1 committed:
Update script.js

- fixed error that does't allow find correct index because tried to find "number" insted "String".

wikitelmex1 committed:
Update script.js

- linters fixed

wikitelmex1 committed:
Update script.js

single quotations instead double

mricanho reviewed (approved):
# Status: ✔️ Project Approved ✔️ 

Hello @peterdgreat  🔥 

You worked hard in all the instructions that the previous Code Reviewers give you👍🏽 

Now you can merge this branch 🚀 

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
